### PR TITLE
Add v2 cells API to chronograf

### PR DESCRIPTION
**Pull request by desa**
_Briefly describe your proposed changes:_

This PR introduces a `cells` API for directly interacting with cells. Cells are now a separate resource from dashboards. This was done under `v2`. This work was done in progression towards a `v2` dashboards API.

TODO
 
  - [x] Add tests for API
  - [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergeable
  - [x] Tests pass
  - [x] swagger.json updated (if modified Go structs or API)

### CODE REVIEW

**Comment by desa** [comment]
@goller just want to get some feedback on the general style before continuing implementation.

The general plan I have is to 

1. Implement Cells
2. Implement Template Variables
3. Implement Dashboards

The goal it to make things usable by the chronograf team to get immediate feedback on the new dashboard API.

File: server/stores.go [hunk 1]
```diff
@@ -106,6 +109,8 @@ type Store struct {
 	MappingsStore      chronograf.MappingsStore
 	OrganizationsStore chronograf.OrganizationsStore
 	ConfigStore        chronograf.ConfigStore
+
```
Comment (goller) [inline server/stores.go]: remove line

**Comment by desa** [comment]
@goller should be ready for another review

File: server/cellsv2.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+type postCellRequest struct {
+	Cell *chronograf.Cell
+}
+
+func decodePostCellRequest(ctx context.Context, r *http.Request) (*postCellRequest, error) {
+	c := &chronograf.Cell{}
+	if err := json.NewDecoder(r.Body).Decode(c); err != nil {
+		return nil, err
+	}
+	return &postCellRequest{
```
Comment (goller) [inline server/cellsv2.go]: What do you think of adding a ValidCell function and test?

File: server/cellsv2.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+type patchCellRequest struct {
+	CellID chronograf.ID
+	Upd    chronograf.CellUpdate
+}
+
+func decodePatchCellRequest(ctx context.Context, r *http.Request) (*patchCellRequest, error) {
+	upd := chronograf.CellUpdate{}
+	if err := json.NewDecoder(r.Body).Decode(&upd); err != nil {
```
Comment (goller) [inline server/cellsv2.go]: Similar to the POST I'd love to see a ValidCellUpdate function and test.

File: server/cellvs2_test.go [hunk 4]
```diff
@@ -0,0 +1,553 @@
...
+	}
+
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		wants  wants
+	}{
+		{
+			name: "get all cells",
```
Comment (goller) [inline server/cellvs2_test.go]: would you add a test for getting cells when there are none?

File: server/cellvs2_test.go [hunk 5]
```diff
@@ -0,0 +1,553 @@
...
+								},
+							}, nil
+						}
+
+						return nil, fmt.Errorf("not found")
+					},
+				},
+			},
+			args: args{
+				id: "2",
```
Comment (goller) [inline server/cellvs2_test.go]: This test doesn't have an update body.  Perhaps add another test that does?

Also, does it make sense to be StatusOK on an update if there is body?

File: server/swagger_v2.yml [hunk 6]
```diff
@@ -0,0 +1,482 @@
+openapi: "3.0.0"
+info:
+  title: Gateway Service
```
Comment (goller) [inline server/swagger_v2.yml]: Chronograf

File: server/swagger_v2.yml [hunk 7]
```diff
@@ -0,0 +1,482 @@
...
+  version: 0.1.0
```
Comment (goller) [inline server/swagger_v2.yml]: This needs to be the version of chronograf and needs to be added to the release procedure.

File: server/swagger_v2.yml [hunk 8]
```diff
@@ -0,0 +1,482 @@
...
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+    get:
+      tags:
+        - Cells
+      summary: Get all cells
+      parameters:
+        - in: query
```
Comment (goller) [inline server/swagger_v2.yml]: Org is not in the implementation. Should it?

File: server/swagger_v2.yml [hunk 9]
```diff
@@ -0,0 +1,482 @@
...
+            application/json:
+              schema:
+                $ref: "#/components/schemas/Error"
+   patch:
+      tags:
+        - Cells
+      summary: Update a single cell
+      requestBody:
+          description: patching of a cell
+          required: true
```
Comment (goller) [inline server/swagger_v2.yml]: Ok, I agree with that the body should be required.  So the test above that didn't check for empty body should error.

File: server/swagger_v2.yml [hunk 10]
```diff
@@ -0,0 +1,482 @@
...
+      summary: Delete a cell
+      parameters:
+        - in: path
+          name: cellID
+          schema:
+            type: string
+          required: true
+          description: ID of cell to update
+      responses:
+        '202':
```
Comment (goller) [inline server/swagger_v2.yml]: The implementation is 204 and I think 204 is more appropriate.

File: v2/cell.go [hunk 11]
```diff
@@ -0,0 +1,313 @@
+package chronograf
```
Comment (goller) [inline v2/cell.go]: What do you think about just calling this platform?  I imagine that'll make it slightly easier when we move it.

File: server/cellsv2.go [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/bouk/httprouter"
+	chronograf "github.com/influxdata/chronograf/v2"
+)
+
+type cellV2Links struct {
```
[the remark this answers is not in the document]
Comment (desa) [reply server/cellsv2.go]: What's this mean?

File: server/cellsv2.go (hunk 2 above)
Comment (desa) [reply server/cellsv2.go]: will do

File: server/cellsv2.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (goller) [reply server/cellsv2.go]: @desa here: https://github.com/influxdata/chronograf/blob/master/server/routes.go#L34

File: server/cellsv2.go [hunk 13]
```diff
@@ -0,0 +1,220 @@
...
+
+	w.WriteHeader(http.StatusNoContent)
+}
+
+type deleteCellRequest struct {
+	CellID chronograf.ID
+}
+
+func decodeDeleteCellRequest(ctx context.Context, r *http.Request) (*deleteCellRequest, error) {
+	param := httprouter.GetParamFromContext(ctx, "id")
```
[the remark this answers is not in the document]
Comment (desa) [reply server/cellsv2.go]: It get's it as an integer and I had the cell ID as a string. Should I change cell ID to be an int?

[the remark this answers is not in the document]
Comment (goller) [reply server/cellsv2.go]: Nah... strike this comment.  When this is moved to platform we won't have those helpers anyway.  In general I don't like integer ids... feels like a leaky abstraction over sequence numbers from yesteryear

File: server/cellsv2.go (hunk 12 above)
[the remark this answers is not in the document]
Comment (desa) [reply server/cellsv2.go]: ah, duh 😞

File: bolt/client.go [hunk 14]
```diff
@@ -167,6 +167,9 @@ func (c *Client) initialize(ctx context.Context) error {
 		if _, err := tx.CreateBucketIfNotExists(OrganizationConfigBucket); err != nil {
 			return err
 		}
+		// Always create Cells bucket.
+		if err := c.initializeCells(ctx, tx); err != nil {
```
[the remark this answers is not in the document]
Comment (desa) [reply bolt/client.go]: heh 😂

File: server/swagger_v2.yml (hunk 7 above)
Comment (desa) [reply server/swagger_v2.yml]: > and needs to be added to the release procedure.

what does this mean?

File: server/swagger_v2.yml (hunk 8 above)
Comment (desa) [reply server/swagger_v2.yml]: nope not yet

File: server/swagger_v2.yml (hunk 7 above)
Comment (goller) [reply server/swagger_v2.yml]: probably to `.bumpversion.cfg` (according to docs/release.md)

@lukevmorris is bumpversion used in the release process to update the release numbers?

**Comment by lukevmorris** [comment]
Yep.

**Comment by desa** [comment]
@goller should be ready for another review.
